src/game/boe.items.cpp

```diff
@@ -894,31 +894,46 @@ short get_num_response(short min, short max, std::string prompt, std::vector<std
 			}
 			return true;
 		});
+
+	if(cancel_value){
+		numPanel["cancel"].attachClickHandler([cancel_value](cDialog& me,std::string,eKeyMod) -> bool {
+			me.setResult<int>(*cancel_value);
+			me.toast(false);
+			return true;
+		});
+	}else{
+		numPanel["cancel"].hide();
+	}
+
 	numPanel.run();
 	
 	return numPanel.getResult<int>();
 }
 
 static bool select_pc_event_filter (cDialog& me, std::string item_hit, eKeyMod) {
 	me.toast(true);
-	if(item_hit != "cancel") {
+	if(item_hit == "pick-all"){
+		me.setResult<short>(7);
```

Comment:
This doesn't seem great. 6 is used nearly universally to mean "the entire party", and _only_ here that meaning is conveyed by 7 instead?

Reply:
I think it would make sense for cancel to be -1 and all to be 6.

I'd just have to carefully change callers that expect 6 to be a cancellation.

Reply:
@CelticMinstrel I set about trying to change this, and I what I found is that the magic numbers and comparisons of return values are so pervasive in the code it feels impossible to change every instance without breaking something. Return values from select_pc() get stored in global state which ends up getting checked in other areas of the code. Lots of variables that get assigned to select_pc() are expecting 6 as their empty value, and getting assigned 6 from other places.

I don't think it's a worthwhile refactor to make.

Reply:
For what it's worth, there are _definitely_ game mechanics that target the entire party, rather than an as-yet-unspecified member of the party.

Reply:
But not ones where it’s a choice between 1 and all, no? I can only think of
the “____ All” spells and there’s no selection dialog for those.